components/experimental/src/dimension/provider/units.rs

```diff
@@ -9,10 +9,78 @@
 //!
 //! Read more about data providers: [`icu_provider`]
 
-
 use alloc::borrow::Cow;
 use icu_provider::prelude::*;
 use tinystr::UnvalidatedTinyAsciiStr;
-use zerovec::{VarZeroVec, ZeroMap};
+use zerovec::{VarZeroVec, ZeroMap2d};
+
+// TODO: implement the units provider
+// thoughts:
+//   use the category (e.g. length, temperature, etc.) as secondary key
+//   have four maps:
+//     - one for the long width patterns (e.g. "{0} meter")
+//     - one for the short width (e.g. "{0} m"?)
+//     - one for the narrow width (e.g. "{0} m")
+//     - one for the DisplayNames (e.g. "meter")
+
+#[icu_provider::data_struct(UnitsDisplayNameV1Marker = "units/displaynames@1")]
```

Comment:
```suggestion
#[icu_provider::data_struct(
    marker(TimeDisplayNameV1Marker = "units/displaynames/time@1"),
    marker(LengthDisplayNameV1Marker = "units/displaynames/length@1"),
)]
```

Reply:
let's discuss the formatter design first, I am going to have a provider that has only time and length data.